### ISSUE TITLE
Handling issues regarding C++ packages

### ISSUE THREAD

**Issue by luoyanjie** [Coursemology/coursemology2#2121]

**Comment by fonglh** [comment]
Exploring the use of [GoogleTest](https://github.com/google/googletest/) for C++ autograding.

**Comment by fonglh** [comment]
GoogleTest provides a `RecordProperty` function which can be used to populate our `expression`, `expected`, `output` and `hint` fields.

https://github.com/google/googletest/blob/master/googletest/docs/AdvancedGuide.md#logging-additional-information

However, this records it as a property of the `testcase` tag. It doesn't look easy to output another tag within the `testcase` tag.

```
<testcase name="Negative" status="run" time="0" classname="FactorialTest" expression="Factorial(-5)">```

One solution is to parse the `testcase` tag for the 4 meta values and use those if it exists.

Maintain backward compatibility with the Python report which uses a separate `meta` tag.

**Comment by fonglh** [comment]
Sample package can be found at https://github.com/Coursemology/evaluator-samples/tree/master/c_factorial

**Comment by fonglh** [comment]
#2151, resolved by #2153, allows the metadata attributes to be populated using GoogleTest.

Check GoogleTest's documentation for examples on how to write the tests.
https://github.com/google/googletest/blob/master/googletest/docs/Primer.md
https://github.com/google/googletest/blob/master/googletest/docs/AdvancedGuide.md

Tutors should be able to figure this out themselves from this point on.
